packages/aws-cdk-lib/aws-eks/lib/service-account.ts

```diff
@@ -91,19 +163,61 @@ export class ServiceAccount extends Construct implements IPrincipal {
       throw RangeError('All namespace names must be valid RFC 1123 DNS labels.');
     }
 
-    /* Add conditions to the role to improve security. This prevents other pods in the same namespace to assume the role.
-    * See documentation: https://docs.aws.amazon.com/eks/latest/userguide/create-service-account-iam-policy-and-role.html
-    */
-    const conditions = new CfnJson(this, 'ConditionJson', {
-      value: {
-        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:aud`]: 'sts.amazonaws.com',
-        [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: `system:serviceaccount:${this.serviceAccountNamespace}:${this.serviceAccountName}`,
-      },
-    });
-    const principal = new OpenIdConnectPrincipal(cluster.openIdConnectProvider).withConditions({
-      StringEquals: conditions,
-    });
-    this.role = new Role(this, 'Role', { assumedBy: principal });
+    let principal: IPrincipal;
+    if (props.identityType !== IdentityType.POD_IDENTITY) {
+      /* Add conditions to the role to improve security. This prevents other pods in the same namespace to assume the role.
+      * See documentation: https://docs.aws.amazon.com/eks/latest/userguide/create-service-account-iam-policy-and-role.html
+      */
+      const conditions = new CfnJson(this, 'ConditionJson', {
+        value: {
+          [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:aud`]: 'sts.amazonaws.com',
+          [`${cluster.openIdConnectProvider.openIdConnectProviderIssuer}:sub`]: `system:serviceaccount:${this.serviceAccountNamespace}:${this.serviceAccountName}`,
+        },
+      });
+      principal = new OpenIdConnectPrincipal(cluster.openIdConnectProvider).withConditions({
+        StringEquals: conditions,
+      });
+
+    } else {
+      /**
+       * Identity type is POD_IDENTITY.
+       * Create a service principal with "Service": "pods.eks.amazonaws.com"
+       * See https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html
+       */
+
+      // EKS Pod Identity does not support Fargate
+      if (cluster instanceof FargateCluster) {
+        throw Error('Pod Identity is not supported in Fargate. Use IRSA identity type instead.');
+      }
+      principal = new ServicePrincipal('pods.eks.amazonaws.com');
```

Comment:
for my education only, is this principle for the trust relationship or it is for other usage?

Reply:
this would be the principal for trust relationship of for the role it auto generates.